fetch/fetch.go

```diff
@@ -189,21 +196,42 @@ type Fetch struct {
 	onlyOnce             sync.Once
 	doneChan             chan struct{}
 	dbLock               sync.RWMutex
+	// FIFO cache of hash->[peer set] to keep track of peers to pull blocks/ballots/txs from
+	hashToPeers      map[types.Hash32]*hashPeers
+	hashToPeersM     sync.RWMutex
+	hashToPeersStats CacheStats
+}
+
+// CacheStats stores FIFO cache statistics.
+type CacheStats struct {
```

Comment:
please move all the cache related stuff to a new file. readers of fetch.go doesn't really need to know how cache works. 

i think for memory management we should do a size-based. can you try to adapt to using *lru.Cache (see AtxCache) and use a size that's adequate based on your systests? it's not FIFO but close enough.

Reply:
> please move all the cache related stuff to a new file. readers of fetch.go doesn't really need to know how cache works.

this will be `CacheStats` moved only as `HashToPeersCacheHit` / `HashToPeersCacheMiss` / `getHashToPeersCacheStats` are `Fetch` methods and probably no point to move object methods partly to a new file, no?

Reply:
no. i mean move the datastructure and CacheStats. take a look of `AtxCache` impl and add `CacheStats` to that. this cache will have API like below
```
TrackHashes([]types.Hash32, p2p.Peer)
Get(types.Hash32)[]p2p.Peer
```

Reply:
or just `Add` and `Get`. i tend to be too verbose

Reply:
done, cheers